arrays/arrays.c

```diff
@@ -83,14 +107,34 @@ char *arr_read(Array *arr, int index) {
 void arr_insert(Array *arr, char *element, int index) {
 
   // Throw an error if the index is greater than the current count
+  if (index > arr->count) {
+    switch(errno) {
+      default:
+        fprintf(stderr, "Inserting into array -> Error: Index is out of bounds \n");
+    }
+
+    exit(1);
+  }
 
   // Resize the array if the number of elements is over capacity
+  if (arr->count + 1 > arr->capacity) {
```

Comment:
This is nice, I like how you check the size before you even deal with inserting into the array.